### PR TITLE
Prakash282 develop 2.2 patch 17465

### PR DESCRIPTION
**Pull request by mageprakash**
### Description
Added proper logic to create and update bundle product option title value store wise.

### Fixed Issues (if relevant)
magento/magento2#17465: Not able to set bundle product option title value store wise in magento 2 #17465

### Manual testing scenarios

On any bundle product, we need to set different store view title as per store.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=17577) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @prakash282. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

File: app/code/Magento/Bundle/Model/ResourceModel/Option.php [hunk 1]
```diff
@@ -81,30 +81,44 @@ protected function _afterSave(\Magento\Framework\Model\AbstractModel $object)
     {
         parent::_afterSave($object);
 
-        $condition = [
+        $conditions = [
             'option_id = ?' => $object->getId(),
-            'store_id = ? OR store_id = 0' => $object->getStoreId(),
+            'store_id = ?' => $object->getStoreId(),
```
Comment (dmanners) [inline app/code/Magento/Bundle/Model/ResourceModel/Option.php:86]: Is there any chance that the `getStoreId` is not set? In this case if it is not set what would happen here?

File: app/code/Magento/Bundle/Model/ResourceModel/Option.php [hunk 2]
```diff
@@ -81,30 +81,44 @@ protected function _afterSave(\Magento\Framework\Model\AbstractModel $object)
...
+                $this->getTable('catalog_product_bundle_option_value'),
+                [
+                    'title' => $object->getTitle()
+                ],
+                $conditions
+            );
+
+            if ($object->getStoreId() != '0')
+                {
+                    $conditions['store_id = ?'] = 0;
```
Comment (dmanners) [inline app/code/Magento/Bundle/Model/ResourceModel/Option.php:103]: If the store id on the object is not 0 why set the conditions store id to be 0?

**Review by dmanners — CHANGES_REQUESTED** [review]
I have some questions raised on this PR. Thanks for your feedback.

File: app/code/Magento/Bundle/Model/ResourceModel/Option.php (hunk 1 above)
Comment (mageprakash) [reply app/code/Magento/Bundle/Model/ResourceModel/Option.php:86]: if we use this 'store_id = ? OR store_id = 0' => $object->getStoreId() then we always get deafult store entry while we update store view title for a particular store. First We checked current store tile exits or not.

File: app/code/Magento/Bundle/Model/ResourceModel/Option.php (hunk 2 above)
Comment (mageprakash) [reply app/code/Magento/Bundle/Model/ResourceModel/Option.php:103]: This code allow to update default_title, while we save store_view title for a particular store.
 
if ($object->getStoreId() != '0')
{
    $conditions['store_id = ?'] = 0;
    $connection->update(
        $this->getTable('catalog_product_bundle_option_value'),
        [
            'title' => $object->getDefaultTitle()
        ],
        $conditions
    );
}

**Comment by orlangur** [comment]
@prakash282 please create PR for `2.3-develop` first.

**Pull request closed without merging**